Scripts/prepare-coverage-reports.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/zsh -l
+set -e
+
+function exportlcov() {
+    build_type=$1
+    executable_name=$2
+
+    executable=$(find "${directory}" -type f -name $executable_name)
+    profile=$(find "${directory}" -type f -name 'Coverage.profdata')
+    output_file_name="$executable_name.lcov"
+
+    can_proceed=true
+    if [[ $build_type == watchOS* ]]; then
+        echo "\tAborting creation of $output_file_name – watchOS not supported."
+    elif [[ -z $profile ]]; then
+        echo "\tAborting creation of $output_file_name – no profile found."
+    elif [[ -z $executable ]]; then
+        echo "\tAborting creation of $output_file_name – no executable found."
+    else
+        output_dir=".build/artifacts/$build_type"
+        mkdir -p $output_dir
+
+        output_file="$output_dir/$output_file_name"
+        echo "\tExporting $output_file"
+        xcrun llvm-cov export -format="lcov" $executable -instr-profile $profile > $output_file
```

Comment:
Trying to make sure I follow: how did we get this report before and why do we need to get it differently now?

Reply:
Before we were using codecov's bash uploader to create this script for us. That script is deprecated. The new system fails to find the `xcresult` files that the bash uploader could find (https://github.com/dfed/CacheAdvance/pull/68#discussion_r992933625), so now we have to create a `lcov` file (which the new system understands and can find) ourselves.

Basically, the new system's Xcode support is lacking, so I created a file type it understood natively.

Reply:
Ah I missed that comment since it was on an outdated line. Thanks for the pointer 👍

Reply:
All good – I should have re-applied that comment!